syntax = "proto2";
option optimize_for = CODE_SIZE;

message debug_data {
   optional uint32     update_period = 1;
   optional uint32     debug_var_1 = 2;
   optional uint32     debug_var_2 = 3;
}

message sample {
    required uint32     config_version = 1;
    required sint32     temperature = 2;
    required uint32     fw_version = 3;
    optional uint32     trigger_code = 4;
    optional uint32     trigger_counter = 5;
    optional sint32     pitch = 6;
    optional sint32     roll = 7;
    optional debug_data data = 8;
}

message trigger_setup{
   optional uint32      code = 1;
   optional uint32      option_1 = 2;
   optional uint32      option_2 = 3;
   optional uint32      option_3 = 4;
}

message set_config {
   required uint32      config_version = 1;
   required uint32      timestamp = 2;
   optional uint32      device_update_period = 3;
   optional uint32      datarate = 4;
   repeated trigger_setup       settings = 8;
}

message get_debug_data {
}